src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -2699,6 +2701,96 @@ CPalThread::GetCachedStackLimit()
     return m_stackLimit;
 }
 
+void
+CPalThread::BeginTrackingSystemCallErrors()
+{
+    _ASSERTE(!m_isTrackingSystemCallErrors);
+
+    m_isTrackingSystemCallErrors = true;
+    m_systemCallErrorsLength = 0;
+    if (m_systemCallErrors != nullptr)
+    {
+        m_systemCallErrors[0] = '\0';
+    }
+}
+
+void
+CPalThread::AppendSystemCallError(
+    LPCSTR format,
+    ...
+    )
+{
+    CPalThread *thread = GetCurrentPalThread();
+    if (thread == nullptr || !thread->m_isTrackingSystemCallErrors)
+    {
+        return;
+    }
+
+    const int BufferSize = 256;
+    char *buffer = thread->m_systemCallErrors;
+    if (buffer == nullptr)
+    {
+        buffer = new(std::nothrow) char[BufferSize];
```

Comment:
The PAL code should not use the new operator other than the placement one. Please use `InternalNew` instead. Using the new operator ends up calling the operator defined in coreclr runtime, which is not expected and can cause crashes (such issue was fixed in https://github.com/dotnet/coreclr/pull/18457).

Reply:
After updating based on other feedback, `new` is not used anymore and a stack-allocated buffer is passed around instead.